### PR TITLE
[Pipelines] Migrate to 1ES pipeline

### PR DESCRIPTION
**Pull request by AnatolyPristensky**
Migrating Pipelines to 1ES Template. 

[AB#104414](https://dev.azure.com/msmobilecenter/Mobile-Center/_boards/board/t/Mobile%20SDK%20team/Backlog%20Items/?workitem=104414)

Successful build: https://dev.azure.com/msmobilecenter/SDK/_build/results?buildId=1520903&view=results

### CODE REVIEW

File: .azurepipelines/build-fastlane-plugin-1es.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            azureSubscription: 'AC - Dev Infra & Build Pool'
+            KeyVaultName: 'mobile-center-sdk'
+            SecretsFilter: 'appcenter-sdk-managed-identity-clientid'
+            RunAsPreJob: false
+            
+        - task: APIScan@2
+          displayName: 'Run APIScan'
+          inputs:
+            softwareFolder: '$(Agent.BuildDirectory)\gem'
+            softwareName: 'appcenter-cli'
```
Comment (DmitriyKirakosyan) [inline .azurepipelines/build-fastlane-plugin-1es.yml:81]: fix pls

Comment (AnatolyPristensky) [reply .azurepipelines/build-fastlane-plugin-1es.yml]: Fixed

File: .azurepipelines/build-fastlane-plugin-1es.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          displayName: 'Use Ruby 3.1.x'
+          inputs:
+            versionSpec: '3.1.x'
+
+        - bash: |
+            gem install bundler
+            bundle install --retry=3 --jobs=4
+            rake build
+          workingDirectory: '$(System.DefaultWorkingDirectory)'
+          displayName: 'Bundle install'
```
Comment (DmitriyKirakosyan) [inline .azurepipelines/build-fastlane-plugin-1es.yml:51]: Better to name step "Rake Build"

Comment (AnatolyPristensky) [reply .azurepipelines/build-fastlane-plugin-1es.yml]: Renamed

File: .azurepipelines/build-fastlane-plugin-1es.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+pr:
+- master
+
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+name: $(Build.SourceBranchName)_$(date:yyyyMMdd)$(rev:.r)
```
Comment (lucen-ms) [inline .azurepipelines/build-fastlane-plugin-1es.yml:13]: Please update name to handle `/` - and issue PRs for all GitHub repos too.

**Review by DmitriyKirakosyan — APPROVED** [review]

**Review by lucen-ms — APPROVED** [review]

**Pull request merged**